### ISSUE TITLE
Starflight 1 & 2 wormhole sound fx

### ISSUE THREAD

**Issue by bsmith1702** [dosbox-staging/dosbox-staging#3278]
### Are you using the latest Dosbox-Staging Version?

- [X] I have checked releases and am using the latest release.

### What Operating System are you using?

Windows 10

### Have you checked that no other similar issue already exists?

- [X] I have searched and not found similar issues.

### A clear and concise description of what the bug is.

I've noticed that the sound as the ship travels through a wormhole in both Starflight 1 and 2 doesn't sound right in either version .80.1 or the RC.  It's just a handful of pops.  The other sound fx in the game seem to be fine.  Other versions of DOSBox seem to have a problem here.  DOSBox-ECE just pops once and then the sound cuts out.  DOSBox-X, however, seems to do a better job.

### Code of Conduct & Contributing Guidelines

- [X] Yes, I agree.

**Comment by kcgen** [comment]
Thanks @bsmith1702 ; can you share the most direct way to reproduce this (to get to that part of the game?)

I remember some of these games have quite cryptic control keys; so any tips or a save game file that puts you right before the wormhole sequence would be very appreciated.

**Comment by johnnovak** [comment]
If not a regression compared to 0.80.1, then we should deal with this in the 0.82.0 period after 0.81.0-RC is out.

**Comment by Grounded0** [comment]
> If not a regression compared to 0.80.1, then we should deal with this in the 0.82.0 period after 0.81.0-RC is out.

Agreed. The RCs fix regressions and aim for a release. Could be fixed in 0.81.x too.

**Comment by bsmith1702** [comment]
> Thanks @bsmith1702 ; can you share the most direct way to reproduce this (to get to that part of the game?)
> 
> I remember some of these games have quite cryptic control keys; so any tips or a save game file that puts you right before the wormhole sequence would be very appreciated.

[starflight.zip](https://github.com/dosbox-staging/dosbox-staging/files/13783422/starflight.zip)

Thanks.

I've attached Starflight 1 with the game saved just outside a wormhole (the shifting stars).  I've included the conf file.  This is the Tandy version of the game so "machine" was set to "tandy".  If you want to find a wormhole quickly yourself, just start a new game with a dummy crew.  All you need is a well-trained navigator.  There's a wormhole nearby at 128x105 in Starflight 1 or 33x70 in Starflight 2.  The ship launching from Starport makes the same noise as well.

**Comment by dani-jozsef** [comment]
I wonder if this might be connected with #3435, another pc speaker issue where only clicks are played

**Comment by johnnovak** [comment]
> and we could benefit from @johnnovak recording what the wormhole\exit spacedock\button sounds should actually sound like

Yeah I'll do that. Added test on hardware label. I'll batch all these hardware tests up then I'll spend half a day recording stuff on hardware, etc.

**Comment by bsmith1702** [comment]
Hi,
    Thank you all for taking an interest in this.  One thing I thought I'd mention but probably isn't of much help was that I tried this with NewRisingSun's TransCopy version of DOSBox.  For whatever reason, the wormhole sound effects sound pretty close to what I remember it sounding, though it has been ages since I've heard it through authentic hardware.  It is interesting that different versions give very different results for this one noise.  But, my understanding is that the source code for the TransCopy version has not been made available, so, like I said, this probably isn't much help.

**Comment by johnnovak** [comment]
> It is interesting that different versions give very different results for this one noise.

The audio device emulations went through several major stages of improvements compared to the original DOSBox code the TC version is supposedly based on. But sometimes we might improve the situation for say 70% of games, but regress things for the 1-2%. That's just how emulator development works.

> But, my understanding is that the source code for the TransCopy version has not been made available, so, like I said, this probably isn't much help.

Indeed 😆 And the original PC speaker code in DOSBox is just an approximation (but it might get a few games "more correct" than the overall-better-but-still-not-quite-perfect improved PC speaker code 😆).
